app/javascript/dashboard/routes/dashboard/settings/campaigns/Campaign.vue

```diff
@@ -58,13 +73,27 @@ export default {
     campaigns() {
       return this.$store.getters['campaigns/getCampaigns'](this.campaignType);
     },
+    displayedCampaigns() {
+      if (this.searchQuery.length > 0) {
+        // Filter campaigns based on partial match in title
+        return this.filteredCampaigns;
+      }
+      // If no search query, return all campaigns
+      return this.campaigns;
+    },
     showEmptyResult() {
       const hasEmptyResults =
         !this.uiFlags.isFetching && this.campaigns.length === 0;
       return hasEmptyResults;
     },
   },
   methods: {
+    searchCampaigns() {
```

Comment:
This assumes that all campaigns are fetched on first load. Can you make sure they are fetched and there's no limit and offset? If theres a limit and offset, we will need to add a server base search